packages/snjs/lib/models/app/search.ts

```diff
@@ -0,0 +1,40 @@
+/** Grouping of required state variables to perform the search */
+
+import { CollectionSort } from "@Lib/protocol/collection/item_collection";
+import { SNTag } from "./tag";
+
+export interface SearchPayload {
+  tag: SNTag,
+  sortBy: CollectionSort,
+  isReversedSort: "asc" | "dsc" | undefined,
+  showArchiveOrTrashed: boolean,
+  hidePinned: boolean,
+  query: string
+
+}
+
+export class Search {
+
+  /**
+   * This is a static class so instantiation should be impossible.
```

Comment:
What do you mean by this is a static class? There's really no such thing. Do you mean abstract class?

Reply:
I mean a poor version of a static class. I know JS doesn't support static classes. This was a hacky workaround.